src/Castle.Core/DynamicProxy/Internal/ClassMethodWithoutTargetInvocation.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright 2004-2021 Castle Project - http://www.castleproject.org/
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Castle.DynamicProxy.Internal
+{
+	using System;
+	using System.ComponentModel;
+	using System.Diagnostics;
+	using System.Reflection;
+
+#if FEATURE_SERIALIZATION
+	[Serializable]
+#endif
+	[EditorBrowsable(EditorBrowsableState.Never)]
+	public sealed class ClassProxyMethodWithoutTargetInvocation : InheritanceInvocation
```

Comment:
I'm probably going to rename this new type to `InheritanceInvocationWithoutTarget`. Far less specific to a single use case than the current name (and thus more likely to be reusable for other purposes), while still perfectly accurate.